vscode/src/debugger/activate.ts

```diff
@@ -125,15 +125,27 @@ class QsDebugConfigProvider implements vscode.DebugConfigurationProvider {
 }
 
 export const workspaceFileAccessor: FileAccessor = {
-  async readFile(uri: string): Promise<Uint8Array> {
-    return await vscode.workspace.fs.readFile(vscode.Uri.parse(uri));
+  normalizePath(path: string): string {
+    return path.replace(/\\/g, "/");
   },
-  async readFileAsString(uri: string): Promise<string> {
-    const contents = await this.readFile(uri);
-    return new TextDecoder().decode(contents);
+  convertToWindowsPathSeparator(path: string): string {
+    return path.replace(/\//g, "\\");
   },
-  async writeFile(uri: string, contents: Uint8Array) {
-    await vscode.workspace.fs.writeFile(vscode.Uri.parse(uri), contents);
+  resolvePathToUri(path: string): vscode.Uri {
+    const normalizedPath = this.normalizePath(path);
+    return vscode.Uri.parse(normalizedPath, false);
+  },
+  async openPath(path: string): Promise<vscode.TextDocument> {
+    const uri: vscode.Uri = this.resolvePathToUri(path);
+    return this.openUri(uri);
+  },
+  async openUri(uri: vscode.Uri): Promise<vscode.TextDocument> {
+    try {
+      return await vscode.workspace.openTextDocument(uri);
+    } catch {
+      const path = this.convertToWindowsPathSeparator(uri.toString());
```

Comment:
It's weird this is needed, as Windows typically handles forward slashes in paths fine as well. What was failing without the conversion?

Reply:
```
[trace] breakpointLocationsRequest: %O {"source":{"name":"BernsteinVazirani.qs","path":"c:\\src\\qsharp\\samples\\algorithms\\BernsteinVazirani.qs"},"line":34}

[error] Failed to open file: CodeExpectedError: cannot open c:/src/qsharp/samples/algorithms/BernsteinVazirani.qs. Detail: Unable to resolve resource c:/src/qsharp/samples/algorithms/BernsteinVazirani.qs
```